pkg/core/resources/apis/meshexternalservice/hostname/generator.go

```diff
@@ -84,18 +85,24 @@ func (g *MeshExternalServiceHostnameGenerator) GenerateHostname(generator *hostn
 		return "", fmt.Errorf("failed compiling gotemplate error=%q", err.Error())
 	}
 	type meshedName struct {
-		Name      string
-		Namespace string
-		Mesh      string
+		Name        string
+		DisplayName string
+		Namespace   string
+		Mesh        string
 	}
 	name := es.GetMeta().GetNameExtensions()[model.K8sNameComponent]
 	if name == "" {
 		name = es.GetMeta().GetName()
 	}
+	displayName, ok := es.GetMeta().GetLabels()[mesh_proto.DisplayName]
+	if !ok {
+		displayName = name
+	}
 	err = tmpl.Execute(&sb, meshedName{
-		Name:      name,
-		Namespace: es.GetMeta().GetNameExtensions()[model.K8sNamespaceComponent],
-		Mesh:      es.GetMeta().GetMesh(),
+		Name:        name,
+		DisplayName: displayName,
+		Namespace:   es.GetMeta().GetNameExtensions()[model.K8sNamespaceComponent],
```

Comment:
Should we just use the label now?

Reply:
Isn't that a label? TBH I don't know why it was `.GetNameExtensions()[model.K8sNamespaceComponent]` and I did not modify that part.

Reply:
I kind of agree. I'd just drop `.Namespace`. We'll recommend label anyways
https://github.com/kumahq/kuma/pull/10478/files

Reply:
That MADR doesn't really address namespace explicitly but I'm putting my question there

Reply:
btw. this extension thing won't work for synced mesh services. This would point to a system namespace

Reply:
merging as is. Discussion moved to PR 10478